torch/csrc/cuda/Module.cpp

```diff
@@ -235,6 +235,60 @@ PyObject * THCPModule_emptyCache(PyObject *_unused, PyObject *noargs)
   Py_RETURN_NONE;
 }
 
+PyObject * THCPModule_memoryEventCounts(PyObject *_unused, PyObject *arg)
+{
+  HANDLE_TH_ERRORS
+  THPUtils_assert(THPUtils_checkLong(arg), "invalid argument to memory_allocated");
+  int device = (int) THPUtils_unpackLong(arg);
+
+  PyObject* result = PyDict_New();
+
+  #define _RESULT_SET_UNSIGNED_LONG(name, val) \  // NOLINT: allow function-like macro
+    do { \  // NOLINT
+      PyDict_SetItemString(result, #name, PyLong_FromUnsignedLongLong(val)); \  // NOLINT
```

Comment:
Will do, thanks

Reply:
Actually, all existing PyObject creation in this file doesn't check for null return value – is there a special reason for that?

Reply:
Some are probably buggy; others may be OK. i.e this is fine:

https://github.com/pytorch/pytorch/blob/b144d057b2211c2773f5175d07a96e3dfeea0b7f/torch/csrc/cuda/Module.cpp#L338

because the return statement propagates the NULL error return.